cli/connect/history.go

```diff
@@ -0,0 +1,133 @@
+package connect
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/tarantool/tt/cli/util"
+)
+
+// commandHistory stores console command history.
+type commandHistory struct {
+	filepath    string
+	commands    []string
+	timestamps  []int64
+	maxCommands int
+}
+
+// parseHistoryCells extracts timestamps and commands from history lines.
+// If format doesn't contain timestamps, it sets the current timestamp
+// for each command.
+func parseHistoryCells(
+	lines []string) (commands []string, timestamps []int64) {
+	regex := regexp.MustCompile(`^#\d+$`)
+
+	// startPos is the first position of a timestamp.
+	startPos := -1
+	for i, line := range lines {
+		if regex.MatchString(line) {
+			startPos = i
+			break
+		}
+	}
+	timestamps = make([]int64, 0)
+	if startPos == -1 {
+		// Read one line per command.
+		// Set the current timestamp for each command.
+		commands = lines
+		timestamp := time.Now().Unix()
+		for range lines {
+			timestamps = append(timestamps, timestamp)
+		}
+		return
+	}
+
+	commands = make([]string, 0)
+	for startPos < len(lines) {
+		j := startPos + 1
+
+		// Move pointer to the next timestamp.
+		for j < len(lines) && !regex.MatchString(lines[j]) {
+			j++
+		}
+
+		// Extract the current timestamp.
+		timestamp, err := strconv.Atoi(lines[startPos][1:])
+
+		if j != startPos+1 && err == nil {
+			timestamps = append(timestamps,
+				int64(timestamp))
+			commands = append(commands,
+				strings.Join(lines[startPos+1:j], "\n"))
+		}
+		startPos = j
+	}
+
+	return
+}
+
+// load loads console history from the history file.
+func (history *commandHistory) load() error {
+	if history == nil {
+		return nil
+	}
+
+	rawLines, err := util.GetLastNLines(history.filepath, history.maxCommands)
```

Comment:
What if we have multiline commands in the history file? Latest 100 lines may be only 10 commands.

Reply:
Yes, that's true. But here i used `maxCommands` as the upper bound. Probably user don't need 10000 commands exactly.

Reply:
> Probably user don't need 10000 commands exactly.
Probably yes.